src/Data/Functor/Foldable/TH.hs

```diff
@@ -70,12 +74,59 @@ import Paths_recursion_schemes (version)
 -- as we don't try to do better than
 -- <https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#deriving-functor-instances GHC's DeriveFunctor>.
 --
-makeBaseFunctor :: Name -> DecsQ
-makeBaseFunctor = makeBaseFunctorWith baseRules
+-- Some doctests:
+--
+-- >>> data Expr a = Lit a | Add (Expr a) (Expr a) | Expr a :* [Expr a]
+-- >>> ; makeBaseFunctor ''Expr
+--
+-- >>> :t AddF
+-- AddF :: r -> r -> ExprF a r
+--
+-- >>> data Rose f a = Rose a (f (Rose f a))
+-- >>> ; makeBaseFunctor [d| instance Functor f => Recursive (Rose f a) |]
```

Comment:
The main documentation for this function (above these doctests) only shows an example of using `makeBaseFunctor` with a `Name` argument, which leaves the reader wondering why one might want to use `makeBaseFunctor` with other sorts of arguments (e.g., `Dec`s). I think the main documentation should be updated to reflect this, with a clear explanation that (1) explains why `makeBaseFunctor` is a type class, and (2) motivates why you'd want each `MakeBaseFunctor` instance.

Reply:
Added.

Reply:
Yet, I'm not sure what suffices as *clear*

Reply:
I think adding the sentence to the effect of "Allowing `makeBaseFunctor` to take both `Name`s and `Dec`s as an argument is why it exists as a method in a type class" would work for me.

Reply:
This still remains to be done, as far as I can tell.

Reply:
This still remains to be done, as far as I can tell.